substrate-node/pallets/pallet-smart-contract/src/mock.rs

```diff
@@ -333,70 +334,53 @@ pub fn new_test_ext() -> sp_io::TestExternalities {
     t
 }
 pub type TransactionCall = pallet_smart_contract::Call<TestRuntime>;
+pub type ExtrinsicResult = Result<PostDispatchInfo, DispatchErrorWithPostInfo>;
 
 #[derive(Default)]
 pub struct PoolState {
     /// A vector of calls that we expect should be executed
-    pub expected_calls: Vec<(TransactionCall, Result<(), ()>)>,
-    pub calls_to_execute: Vec<(TransactionCall, Result<(), ()>)>,
+    pub expected_calls: Vec<(TransactionCall, ExtrinsicResult)>,
+    pub calls_to_execute: Vec<(TransactionCall, ExtrinsicResult)>,
     pub i: usize,
 }
 
 impl PoolState {
     pub fn should_call_bill_contract(
         &mut self,
         contract_id: u64,
-        block_number: BlockNumber,
-        expected_result: Result<(), ()>,
+        expected_result: ExtrinsicResult,
     ) {
         self.expected_calls.push((
-            crate::Call::bill_contract_for_block {
-                contract_id,
-                block_number,
-            },
+            crate::Call::bill_contract_for_block { contract_id },
```

Comment:
Is it ok if I create an issue for adding the extra argument block number like we talked about? So that we can speficy that the call has to happen at a specific block number.

Reply:
It's done

Reply:
Nice :1st_place_medal: